kotlinpoet/src/main/java/com/squareup/kotlinpoet/FunSpec.kt

```diff
@@ -378,26 +378,52 @@ class FunSpec private constructor(
       parameters += parameterSpec
     }
 
+    @JvmName("callThisConstructorWithStrings")
+    fun callThisConstructor(args: Iterable<String>) = apply {
+      callConstructor("this", args.map { CodeBlock.of(it) })
+    }
+
+    fun callThisConstructor(args: List<CodeBlock>) = apply {
+      callConstructor("this", args)
+    }
+
+    fun callThisConstructor(args: Iterable<CodeBlock>) = apply {
+      callConstructor("this", args.toList())
+    }
+
     fun callThisConstructor(vararg args: String) = apply {
-      callConstructor("this", *args.map { CodeBlock.of(it) }.toTypedArray())
+      callConstructor("this", args.map { CodeBlock.of(it) })
     }
 
     fun callThisConstructor(vararg args: CodeBlock = emptyArray()) = apply {
```

Comment:
It's interesting that we use an `emptyArray()` default arg here but not in the `vararg args: String` overload. This is also missing a `@JvmOverloads`. Out of scope of this PR, so no need to change.

Reply:
should I keep `private fun callConstructor(constructor: String, vararg args: CodeBlock)`? I think it makes duplicated code with `private fun callConstructor(constructor: String, vararg args: CodeBlock)` and `private fun callConstructor(constructor: String, args: List<CodeBlock>)`

Reply:
No, I'm fine with the change you made. The arguments are ultimately kept as a list, so it doesn't matter at which point we convert a `vararg` to a list.